Assets/Scripts/Minion/Controllers/MinionMoveController.cs

```diff
@@ -7,30 +9,65 @@ namespace Minion.Controllers
     public class MinionMoveController : MinionController
     {
         [SerializeField] private MinionSO minionConfig;
+        [SerializeField] private Vector3EventChannelSO onPlayerMovedEvent;
+        [SerializeField] private Vector3EventChannelSO onDashMovementEvent;
+        
+        private Vector3 _moveDir;
 
+        public void OnDisable()
+        {
+            onPlayerMovedEvent.onVectorEvent.RemoveListener(HandleMove);
+            onDashMovementEvent.onVectorEvent.RemoveListener(HandleMove);
+        }
 
-        private Vector3 _moveDir;
-        
         public void Enter()
         {
             StartCoroutine(HandleChangeState());
         }
 
         private IEnumerator HandleChangeState()
         {
+            bool isTargetFar = Vector3.Distance(transform.position, target.transform.position) >
+                               minionConfig.moveData.minDistance;
+
+            if (isTargetFar)
+            {
+                _moveDir = target.transform.position - transform.position;
+                _moveDir.y = 0;
+                while (Vector3.Distance(transform.position, target.transform.position) > minionConfig.moveData.minDistance)
+                {
+                    transform.Translate(_moveDir * (minionConfig.moveData.speed * Time.deltaTime));
+                    yield return null;
+                }
+            }
+            else
+            {
+                _moveDir = transform.position - target.transform.position;
+                _moveDir.y = 0;
+                while (Vector3.Distance(transform.position, target.transform.position) < minionConfig.moveData.minDistance)
+                {
+                    transform.Translate(_moveDir * (minionConfig.moveData.speed * Time.deltaTime));
+                    yield return null;
+                }
+            }
+
+            onPlayerMovedEvent.onVectorEvent.AddListener(HandleMove);
+            onDashMovementEvent.onVectorEvent.AddListener(HandleMove);
             yield return new WaitForSeconds(minionConfig.GetRandomMoveTime());
+            onPlayerMovedEvent.onVectorEvent.RemoveListener(HandleMove);
+            onDashMovementEvent.onVectorEvent.RemoveListener(HandleMove);
             MinionAgent.ChangeStateToChargeAttack();
         }
 
+        private void HandleMove(Vector3 movement)
+        {
+            movement.y = 0;
+            transform.position += movement;
```

Comment:
This logic has the problem of not taking into consideration map bounds.
We'll need to update and fix this later on.